### DISCUSSION TITLE
ValueError: conflicting sizes for dimension 'time': length 500 on the data but length 10 on coordinate 'time'

### DISCUSSION THREAD

**Discussion by shv75** [corteva/rioxarray#771] [Q&A]
I am trying to open a netCDF file with python and rioxarray:
xr = rioxarray.open_rasterio(downl_directory+"/cmems_mod_glo_bgc-pft_anfc_0.25deg_P1D-m.nc")

 but it returns this error :
 ValueError: conflicting sizes for dimension 'time': length 500 on the data but length 10 on coordinate 'time'
 
 I have checked the time dimensions of the data and metadata and they match. What can I do to fix it?
 
 Thanks in advance!
[cmems_mod_glo_bgc-pft_anfc_0.25deg_P1D-m.zip](https://github.com/corteva/rioxarray/files/15039246/cmems_mod_glo_bgc-pft_anfc_0.25deg_P1D-m.zip)

**Comment by snowman2** [comment]
If you open your dataset with xarray:

```python
import xarray

xds = xarray.open_dataset("cmems_mod_glo_bgc-pft_anfc_0.25deg_P1D-m.nc", decode_coords="all")
```
The contents of `xds`:
```
<xarray.Dataset>
Dimensions:    (depth: 50, latitude: 41, longitude: 41, time: 10)
Coordinates:
  * depth      (depth) float32 0.494 1.541 2.646 ... 5.275e+03 5.728e+03
  * latitude   (latitude) float32 10.0 10.25 10.5 10.75 ... 19.5 19.75 20.0
  * longitude  (longitude) float32 40.0 40.25 40.5 40.75 ... 49.5 49.75 50.0
  * time       (time) datetime64[ns] 2022-01-01 2022-01-02 ... 2022-01-10
Data variables:
    chl        (time, depth, latitude, longitude) float32 ...
```

That means this is a 4D dataset. For more information about the issue, see: https://github.com/corteva/rioxarray/issues/174

**Comment by ForestMike1** [comment]
Not sure it's a 4D problem.

I ran into the same problem after creating a new environment. In my case, it was .nc4 files that were opening correctly with rioxarray.open_rasterio.

I found that, rioxarray uses rasterio, which offer a gdal driver option, and I didn't had the proper "libgdal-netcdf" install in my new env , while I had it in my old.

Try this:

import rioxarray as rxr 
xds_rio= rxr.open_rasterio('data.grib', driver= [find the good driver in the list below])

GDAL driver: https://gdal.org/en/stable/drivers/raster/index.html

The good driver is obviously indicated with your file extension, but the error message sent if you don't have it wiill indicate the good command to install it (in my case : "conda install -c conda-forge libgdal-netcdf")
